sdk/python/kfp/_client.py

```diff
@@ -452,14 +452,26 @@ def get_experiment(self, experiment_id=None, experiment_name=None, namespace=Non
       raise ValueError('Either experiment_id or experiment_name is required')
     if experiment_id is not None:
       return self._experiment_api.get_experiment(id=experiment_id)
-    next_page_token = ''
-    while next_page_token is not None:
-      list_experiments_response = self.list_experiments(page_size=100, page_token=next_page_token, namespace=namespace)
-      next_page_token = list_experiments_response.next_page_token
-      for experiment in list_experiments_response.experiments or []:
-        if experiment.name == experiment_name:
-          return self._experiment_api.get_experiment(id=experiment.id)
-    raise ValueError('No experiment is found with name {}.'.format(experiment_name))
+    experiment_filter = json.dumps({ 
+        "predicates": [ 
+          { 
+            "op":  _FILTER_OPERATIONS["EQUALS"], 
+            "key": "name", 
+            "stringValue": experiment_name, 
+          },
+          {
+            "op":  _FILTER_OPERATIONS["EQUALS"], 
+            "key": "namespace", 
+            "stringValue": namespace, 
```

Comment:
Can I confirm whether you tested in KFP standalone mode? I don't remember what namespace should it be, is it empty or kubeflow?

Reply:
I didn't realize we could filter namespace like this. However, I think it's more canonical filtering namespace by resource ref.
https://github.com/kubeflow/pipelines/blob/2b95c002b741fadb6ddd2aa364c6a5b55ea78edd/backend/api/python_http_client/kfp_server_api/api/experiment_service_api.py#L548

A rules of thumb is that for any resource filter in resource ref, we suggest using resource ref to filter it, including pipeline, version, namespace, experiment, job...

Reply:
>
> Can I confirm whether you tested in KFP standalone mode?  don't remember what namespace should it be, is it empty or kubeflow?

yes, it should be empty to work.  For singel user mode, it should be empty, if set it will raise an error. . 

>
> ... However, I think it's more canonical filtering namespace by resource ref. 

I will update.